app/controllers/api/v1/search_controller.rb

```diff
@@ -665,22 +665,30 @@ def self.match_results_by_filter(search_result:, result_key:, facets:)
       # execute a search in TDR and get back normalized results
       def self.get_tdr_results(selected_facets:, terms:)
         results = {}
-        if selected_facets.present?
-          facet_json = ApplicationController.data_repo_client.generate_query_from_facets(selected_facets)
-        end
-        if terms.present?
-          term_json = ApplicationController.data_repo_client.generate_query_from_keywords(terms)
-        end
-        # now we merge the two queries together to perform a single search request
-        query_json = ApplicationController.data_repo_client.merge_query_json(facet_query: facet_json, term_query: term_json)
-        logger.info "Executing TDR query with: #{query_json}"
-        raw_tdr_results = ApplicationController.data_repo_client.query_snapshot_indexes(query_json)
-        added_file_ids = {}
-        raw_tdr_results['result'].each do |result_row|
-          results = process_tdr_result_row(result_row, results,
-            selected_facets: selected_facets,
-            terms: terms,
-            added_file_ids: added_file_ids)
+        begin
+          if selected_facets.present?
+            facet_json = ApplicationController.data_repo_client.generate_query_from_facets(selected_facets)
```

Comment:
For succinct readability, could we do something like:

`tdr_client = ApplicationController.data_repo_client` 

atop this controller, then replace references throughout this file accordingly?  E.g.:

```suggestion
            facet_json = tdr_client.generate_query_from_facets(selected_facets)
```

Reply:
This would be a departure from how we reference this API client vs. all the other ones (`FireCloudClient`, `BigQueryClient`, and `PapiClient`).  Furthermore, if we assigned it to a local variable like this, it would mean we're reinstantiating the client for every request, which will reissue OAuth tokens every time.  Eventually we'd run afoul of Google's token quotas.  This is the main reason why we consolidated into class variables in `ApplicationController`.

Reply:
Thanks for the reminder!

Reply:
thanks for the question/context !